app/scripts/BedLikeTrack.js

```diff
@@ -48,57 +62,62 @@ class BedLikeTrack extends HorizontalTiled1DPixiTrack {
     tile.graphics.addChild(tile.rectGraphics);
     tile.graphics.addChild(tile.textGraphics);
 
+    let plusStrandRows = [];
+    let minusStrandRows = [];
 
     if (tile.tileData && tile.tileData.length) {
       tile.tileData.sort((a, b) => b.importance - a.importance);
       // tile.tileData = tile.tileData.slice(0, MAX_TILE_ENTRIES);
 
-      let rows = [];
       if (!this.options || !this.options.valueColumn) {
+        // no value column so we can break entries up into separate
+        // plus and minus strand segments
         const segments = tile.tileData.map((x) => {
           const chrOffset = +x.chrOffset;
 
           return {
             from: +x.fields[1] + chrOffset,
             to: +x.fields[2] + chrOffset,
             value: x,
+            text: x.fields[3],
+            strand: x.fields.length >= 6 && x.fields[5] === '-' ? '-' : '+',
           };
         });
 
-        rows = segmentsToRows(segments);
-        // console.log('rows', rows);
+        plusStrandRows = segmentsToRows(segments.filter(x => x.strand === '+'));
+        minusStrandRows = segmentsToRows(segments.filter(x => x.strand === '-'));
       } else {
-        rows = [tile.tileData.map(x => ({ value: x }))];
+        plusStrandRows = [tile.tileData.map(x => ({ value: x }))];
       }
 
-      tile.rows = rows;
+      tile.plusStrandRows = plusStrandRows;
+      tile.minusStrandRows = minusStrandRows;
 
-      tile.tileData.forEach((td, i) => {
-        const geneInfo = td.fields;
-        const fill = this.options.fillColor ? this.options.fillColor : 'blue';
+      if (this.options.showTexts) {
+        tile.tileData.forEach((td, i) => {
+          const geneInfo = td.fields;
 
-        tile.textWidths = {};
+          tile.textWidths = {};
+          tile.textHeights = {};
 
-        // don't draw texts for the latter entries in the tile
-        if (i >= MAX_TEXTS) {
-          return;
-        }
+          // don't draw texts for the latter entries in the tile
+          if (i >= MAX_TEXTS) {
+            return;
+          }
 
-        // geneInfo[3] is the gene symbol
-        const text = new PIXI.Text(geneInfo[3], {
-          fontSize: this.textFontSize,
-          fontFamily: this.textFontFamily,
-          fill: colorToHex(fill)
-        });
-        if (this.flipText) { text.scale.x = -1; }
+          // geneInfo[3] is the gene symbol
+          const text = new PIXI.Text(geneInfo[3], TEXT_STYLE);
+          if (this.flipText) { text.scale.x = -1; }
 
-        text.anchor.x = 0.5;
-        text.anchor.y = 1;
+          text.anchor.x = 0.5;
+          text.anchor.y = 0.5;
 
-        tile.texts[geneInfo[3]] = text; // index by geneName
+          tile.texts[td.uid] = text; // index by geneName
```

Comment:
Just a sanity check comment: I assume the change from `geneInfo[3]` to `td.uid` is expected? :)

Reply:
Yup, expected. Using `geneInfo[3]` (which corresponds to gene name) can lead to lots of names not being rendered due to segments having the same name.